tests/integration/test_average.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+from utils.comparisons import allclose
+
+import cunumeric as num
+
+axes = [None, 0, 1, 2, (0, 1, 2)]
+
+
+input = [[[12, 3, 1, 2], [9, 1, 6, 1]], [[7, 9, 11, 50], [31, 5, 3, 2]]]
+in_num = num.array(input)
+in_np = np.array(input)
+
+
+@pytest.mark.parametrize("axis", axes)
+def test_no_mask(axis):
+    out_num, scl_num = num.average(in_num, axis=axis, returned=True)
+    out_np, scl_np = np.average(in_np, axis=axis, returned=True)
+    assert allclose(out_num, out_np)
+    assert allclose(scl_num, scl_np)
+
+
+@pytest.mark.parametrize("axis", axes)
+def test_full_weights(axis):
+    weight_input = [[[1, 2, 3, 4], [3, 3, 7, 1]], [[2, 2, 3, 3], [4, 1, 0, 1]]]
+    weights_np = np.array(weight_input)
+    weights_num = num.array(weight_input)
+
+    out_num, scl_num = num.average(
+        in_num, weights=weights_num, axis=axis, returned=True
+    )
+    out_np, scl_np = np.average(
+        in_np, weights=weights_np, axis=axis, returned=True
+    )
+    assert allclose(out_num, out_np)
+    assert allclose(scl_num, scl_np)
+
+
+single_dimension_weights = [
+    [3, 4],
+    [1, 2],
+    [4, 1, 2, 1],
+]
+single_dimension_axis = [0, 1, 2]
+
+
+@pytest.mark.parametrize(
+    "weights,axis", zip(single_dimension_weights, single_dimension_axis)
+)
+def test_single_axis_weights(weights, axis):
+    weights_np = np.array(weights)
+    weights_num = num.array(weights)
+
+    out_num, scl_num = num.average(
+        in_num, weights=weights_num, axis=axis, returned=True
+    )
+    out_np, scl_np = np.average(
+        in_np, weights=weights_np, axis=axis, returned=True
+    )
+    assert allclose(out_num, out_np)
+    assert allclose(scl_num, scl_np)
+
+
```

Comment:
Need tests for the "bad" cases -- confirm expected exception types are raised on specific bad inputs and that the exception type matches the numpy exception type (when appropriate, sometimes there are explicit differences to note, then just test the cunumeric expectation). Here's one example to emulate: 

https://github.com/nv-legate/cunumeric/blob/fbdfe89f5e6783c191c5e5c72c0f476f352d360b/tests/integration/test_put.py#L195-L204

Reply:
Also looks like `returned=False` case needs tests

Reply:
Thank you for the feedback! Added to the PR. Also, would you happen to know why [this](https://github.com/nv-legate/cunumeric/actions/runs/7921873820/job/21628937347?pr=1124) part of the build was failing int he previous commit? It seemed to be unrelated to my changes.